connecting to a miner: port forward with ssh (#14)
darkforest-eth/developer-guides: mining/connecting-to-a-remote-headless-miner.md

# Connecting to a remote \(headless\) miner

> Connecting to a remote explorer should be as simple as putting your miners url in the built in Remote Miner plugin. However if that remote explorer isnt running on your localhost machine, your browser security will reject the connection. The explorer guides explain you nave to put an ssl cert on your remote server (difficult, costly) or disable insecure content for the zkga.me domain (not good security practices in crypto where we really need good security practices). But that isn't the only solution!. Instead we can make a remote server look like a local instance.

Reviewer: ```suggestion
Connecting to a remote explorer should be as simple as putting your miners url in the built in Remote Miner plugin. However if that remote explorer isnt running on your localhost machine, your browser security will reject the connection. The explorer guide gives 2 options to fix this browser limitation. 1) you can put an ssl cert on your remote server (difficult, costly) or 2) disable insecure content for the zkga.me domain (not good security practice in crypto where we really need good security practices). But those aren't the only solution!. Instead we can make a remote server look like a local server with a few command lines.
```

## Port forward with ssh

SSH is a way to access remote machines and allows port forwarding.
To set this up there are a few simple steps:

### 1. Installing ssh

SSH is already installed on GNU/Linux.
For Windows install your favorite application.  

### 2. Editing `~/.ssh/config`

This step is done so that you can ssh into machines without typing commandflags each time. Replace the 192 ip and user with your server.

Here is an example entry:
    
    Host miner
        Hostname 192.168.178.132
        User user
        Port 22
        Localforward 8001 localhost:8000

> If you have multiple miner copy/paste the entry and then increment the first port number by one

### Connecting

    ssh miner

Once you drop into the shell start `mimc-fast` or whatever remote miner client you're using
> Notice how `miner` stands for the entry in the `~/.ssh/config`

### Plugin use

> The Plugin is called Remote Explorer and is in the official client.

Reviewer: ```suggestion
The Plugin is called Remote Explorer and is in the official client, but does need one change from the default```

# Connecting to a remote \(headless\) miner

> To connect to a miner and not having to turn off the browsers SSL/TLS you can:

Reviewer: ```suggestion
To connect to a remote miner (setup not covered in this guide) without having to turn off the browsers SSL/TLS you can:
```

> - issue a Certificate with a Domain you own and setup a proxy (comlicated)

Reviewer: ```suggestion
```

- port worward the miner to your localhost

## Port forward with ssh

SSH is a way to access remote machines and allows port forwarding.
To set this up there are a few simple steps:

### 1. Installing ssh

SSH is already installed on GNU/Linux.
For Windows install your favorite application.  

### 2. Editing `~/.ssh/config`

> This step is done so that you can ssh into machines without typing commandflags each time.

Reviewer: ```suggestion
This step is done so that you can ssh into machines without typing commandflags each time. Replace the hostname,  user, and `localhost:8000` with your port 
```

Here is an example entry:
    
    Host miner
        Hostname 192.168.178.132
        User user
        Port 22
        Localforward 8001 localhost:8000

> If you have multiple miner copy/paste the entry and then increment the first port number by one

### Connecting

    ssh miner

> Once you drop into the shell start `mimc-fast`

Reviewer: ```suggestion
Once you drop into the shell start `mimc-fast` or whatever remote miner client you're using
```
